Polyhedron/demo/Polyhedron/include/CGAL/IO/read_surf_trianglemesh.h

```diff
@@ -0,0 +1,186 @@
+#include <iostream>
+#include <string>
+#include <CGAL/boost/graph/Euler_operations.h>
+#include <CGAL/Polygon_mesh_processing/internal/named_function_params.h>
+#include <CGAL/Polygon_mesh_processing/internal/named_params_helper.h>
+
+//a material is composed of an material Id and a material name.
+typedef std::pair<int, std::string> material;
+
+struct MaterialData
+{
+  material innerRegion;
+  material outerRegion;
+};
+
+/*!
+ * read_surf reads a file which extension is .surf and fills `output` with one `Mesh` per patch.
+ * Mesh is a model of FaceListGraph.
+ */
+template<class Mesh, class NamedParameters>
+void read_surf(std::istream& input, std::vector<Mesh>& output, std::vector<MaterialData>& metadata,  const NamedParameters&)
```

Comment:
@sloriot @afabri 
Do you know a documented method from CGAL that is supposed to create a sequence of polyhedra. Output iterators are a general tool to build an API that generate things, but when the objects are big, that is not efficient. Here I have chosen to hard-code a non-const reference to a `std::vector<Mesh>` but that is not pretty.